scripts/packaging/publish_release.py

```diff
@@ -120,7 +124,10 @@
                             remainingTrials = 0
                         except requests.exceptions.ConnectionError:
                             remainingTrials -= 1
-                            print('Release upload failed (remaining trials: %d)' % remainingTrials)
+                            print('Release upload failed due to connection error (remaining trials: %d)' % remainingTrials)
+                        except requests.exceptions.HTTPError:
+                            remainingTrials -= 1
+                            print('Release upload failed due to server error (remaining trials: %d)' % remainingTrials)
                     if (releaseExists and tagName.startswith('nightly_') and not releaseCommentModified and
```

Comment:
I think we can afford a short delay here, for example, 1 minute, then 2 minutes, then 3 minutes... This way we will be more robust GitHub failures - if a server is unavailable it is likely it will be unavailable in 2-3 seconds as well.

Reply:
I will add a timeout.
Note that i expect that this custom retry is never called because handled before by the Github library internal retry.